### PR TITLE
Update event API and reference docs

### LINKED ISSUE #1078: Document occurrences attributes in Sensu Go
Document the behavior of the following attributes in Sensu Go:

- [ ] `occurrences_watermark`
- [ ] `occurrences`

### LINKED ISSUE #1257: Update events API doc POST and PUT endpoints
Page: https://docs.sensu.io/sensu-go/5.2/api/events

To do:

- [ ] Be more specific about which attributes are required
- [ ] Provide an example for the POST endpoint
- [ ] Give an example of useful attributes to include, like status

Minimum required:

```
{
  "entity": {
    "entity_class": "agent",
    "metadata": {
      "name": "sensu-go-sandbox",
      "namespace": "default"
    }
  },
  "check": {
    "interval": 60,
    "metadata": {
      "name": "check-cpu"
    }
  }
}
```

### LINKED ISSUE #1326: Update event timestamp docs
Events that are sent to the store with a zero value as a timestamp
will now be given the current time instead.

Reference: https://github.com/sensu/sensu-go/pull/2783

### PR DESCRIPTION
**Pull request by apaskulin**
## Description

- Adds event attributes to the events reference (Closes #1078)
- Adds payload parameter details and request examples to the event API docs (Closes #1257)
- Removes requirement for including a timestamp when creating an event via the API (Closes #1326)

## To do
- [x] Copy to all versions
- [x] Include note about event timestamps for versions 5.3 and earlier

### CODE REVIEW

File: content/sensu-go/5.4/reference/events.md [hunk 1]
```diff
@@ -50,267 +63,788 @@ Metric Format. Both the check results, and resulting (extracted) metrics are
...
+example      | {{< highlight shell >}}
+"output": "sensu-go-sandbox.curl_timings.time_total 0.005"
+{{< /highlight >}}
+
+state         |      |
+-------------|------
+description  | The state of the check: `passing`, `failing`, or `flapping` (a rapid change in check result status)
+required     | false
+type         | String
+example      | {{< highlight shell >}}"state": "passing"{{< /highlight >}}
```
Comment (apaskulin) [inline content/sensu-go/5.4/reference/events.md:564]: Is there a relationship between `state` and `status`?

Comment (palourde) [reply content/sensu-go/5.4/reference/events.md:564]: It depends; we [first check](https://github.com/sensu/sensu-go/blob/f5b8db4d5d75c9ed9960ff6d5ed5bd905a3f9812/backend/eventd/flapping.go#L31) if the check is flapping using the check `TotalStateChange`, `LowFlapThreshold` & `HighFlapThreshold`. If not, then [we use the `status`](https://github.com/sensu/sensu-go/blob/f5b8db4d5d75c9ed9960ff6d5ed5bd905a3f9812/backend/eventd/flapping.go#L33) to determine whether is passing or failing.

**Review by palourde — APPROVED** [review]

File: content/sensu-go/5.4/reference/events.md (hunk 1 above)
Comment (apaskulin) [reply content/sensu-go/5.4/reference/events.md:564]: Excellent, I added a note in https://github.com/sensu/sensu-docs/pull/1327/commits/a44a2415e6114bb13804c5fc8fda5a4900e4c394 and https://github.com/sensu/sensu-docs/pull/1327/commits/390c847fe1188ce0d568fc26e8ce7847eb0d093e

Comment (palourde) [reply content/sensu-go/5.4/reference/events.md:564]: 👌

**Pull request merged**